specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2019-08-01/cosmos-db.json

```diff
@@ -93,22 +93,22 @@
             "in": "body",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/DatabaseAccountPatchParameters"
+              "$ref": "#/definitions/DatabaseAccountUpdateParameters"
             },
-            "description": "The tags parameter to patch for the current database account."
+            "description": "The parameters to provide for the current database account."
           }
         ],
         "responses": {
           "200": {
-            "description": "The properties of the database account were patched successfully.",
+            "description": "The database account update operation will complete asynchronously.",
             "schema": {
               "$ref": "#/definitions/DatabaseAccountGetResults"
             }
           }
         }
       },
       "put": {
-        "operationId": "DatabaseAccounts_CreateOrUpdate",
+        "operationId": "DatabaseAccounts_Create",
```

Comment:
PUT should also allow updating the resource. I don't understand why you want to make this change? PUT is meant to be create or Update. Can you please clarify if I am missing something?

Reply:
This method is still create or update, but we want to discourage users from updating resources in this manner since it is a complete replacement. We have had instances where they accidentally disabled a feature because they thought they were performing a PATCH and did not include all properties in the "CreateOrUpdate" request.

This also follows the pattern that [storage](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2019-04-01/storage.json#L126) uses in their swagger. I can change it back to "CreateOrUpdate"

Reply:
@ArcturusZhang Can you comment on any naming conventions we should follow for put and patch operations? Is it ok to name the PUT operation "create"?

Reply:
IMO, lets be consistent with majority of the services in Azure which specify createOrUpdate.